src/actions/pubSubHandlers.js

```diff
@@ -92,14 +92,18 @@ export const handleNewMessage = message => (dispatch, getState) => {
   })
 }
 
-export function handleRemovedMessage({ id }) {
+export function handleRemovedMessage({ id, channel }) {
   return dispatch => {
     dispatch(removeSharedFiles(id))
     dispatch(removeMention(id))
     dispatch({
       type: types.REMOVE_MESSAGE,
       payload: id,
     })
+    dispatch({
+      type: types.SUBTRACT_UNREAD_MESSAGE_COUNTER,
```

Comment:
This is problematic from an architectural point of view. `REMOVE_MESSAGE` is already dispatching and indicating what's happening. The reducers themselves should decide what to do with that information.

By introducing `SUBTRACT_UNREAD_MESSAGE_COUNTER` you imply a direct command that should happen inside a specific reducer. My suggestion would be to remove `SUBTRACT_UNREAD_MESSAGE_COUNTER` completely and update that `REMOVE_MESSAGE` payload to contain the message id and the channel.